src/preamble.js

```diff
@@ -843,7 +843,15 @@ var splitModuleProxyHandler = {
       var imports = {'primary': Module['asm']};
       // Replace '.wasm' suffix with '.deferred.wasm'.
       var deferred = wasmBinaryFile.slice(0, -5) + '.deferred.wasm'
-      instantiateSync(deferred, imports);
+      var mmhSuccess = false;
+      if (Module["mmh"]) {
```

Comment:
Maybe give this better name that is linked to splitModule?  At least perhaps spell it out "methodMissingHandler"?   And perhaps just return the success boolean?  There doesn't seem to be any reason to return the other things (at least not right now).

Reply:
How about `loadSplitModule` for the name?

Also, failure to load the module is not usually recoverable, so instead of falling back to the `instantiateSync` code path if the custom handler fails, let's just throw a hard error. The `instantiateSync` code path can just be the default handler when the user doesn't provide one.

Reply:
sure i can do that